test/apps/iodemo/ucx_wrapper.cc

```diff
@@ -118,10 +120,84 @@ void UcxContext::UcxDisconnectCallback::operator()(ucs_status_t status)
     delete this;
 }
 
-UcxContext::UcxContext(size_t iomsg_size, double connect_timeout, bool use_am) :
-    _context(NULL), _worker(NULL), _listener(NULL), _iomsg_recv_request(NULL),
-    _iomsg_buffer(iomsg_size, '\0'), _connect_timeout(connect_timeout),
-    _use_am(use_am)
+UcxContext::UcpWorkEpoll::UcpWorkEpoll(uint32_t max_events, bool edge_trigger) :
+    _epoll_fd(0),
+    _max_events(max_events),
+    _epoll_events(NULL),
+    _edge_trigger(edge_trigger)
+{
+    _epoll_fd = epoll_create1(EPOLL_CLOEXEC);
+    assert(_epoll_fd > 0);
+
+    _epoll_events = new (std::nothrow) epoll_event[max_events]();
+    assert(_epoll_events != NULL);
+}
+
+UcxContext::UcpWorkEpoll::~UcpWorkEpoll()
+{
+    delete[] _epoll_events;
+    _epoll_events = NULL;
+
+    if (_epoll_fd > 0) {
+        close(_epoll_fd);
+        _epoll_fd = 0;
+    }
+}
+
+void UcxContext::UcpWorkEpoll::add_event(int mon_fd, uint32_t event)
+{
+    ctrl_event(mon_fd, event, EPOLL_CTL_ADD);
+}
+
+void UcxContext::UcpWorkEpoll::del_event(int mon_fd, uint32_t event)
+{
+    ctrl_event(mon_fd, event, EPOLL_CTL_DEL);
+}
+
+void UcxContext::UcpWorkEpoll::mod_event(int mon_fd, uint32_t event)
+{
+    ctrl_event(mon_fd, event, EPOLL_CTL_MOD);
+}
+
+int UcxContext::UcpWorkEpoll::wait_event(int32_t millseconds)
+{
+    return epoll_wait(_epoll_fd, _epoll_events, _max_events, millseconds);
```

Comment:
Why do we need to set _max_events? Can we just set it as 1?

Reply:
If 1 is working, we should hardcode 1 here and remove all max_events related code.

Reply:
Done